RELEASE_GUIDE.md

```diff
@@ -0,0 +1,115 @@
+
+### Feature Branches
+
+Do not creating branches in the upstream repo, use your fork, for the exception of long lasting feature branches that require active collaboration from multiple developers. Name feature branches `feature/<thing>`. Once the work is merged to `main`, please make sure to delete the feature branch.
+
+## Release Labels
+
+Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v2.0.0`, as well as `patch` and `backport`. Use release labels to target an issue or a PR for a given release. See [MAINTAINERS](MAINTAINERS.md#triage-open-issues) for more information on triaging issues.
+
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+## Prerequisites
+
+* Since releases are only done on Thursdays, maintainers should ensure all changes are merged by Tuesday.
+    * A week prior to the scheduled release, maintainers should announce a code freeze in the [#performance channel](https://opensearch.slack.com/archives/C0516H8EJ7R) within the OpenSearch Slack community. See the following example for what that might look like:
+```
+OpenSearch Benchmark release is scheduled for 1/25 and a code freeze will be put in place starting on 1/23.
+```
+* Ensure that version.txt matches the new release version before proceeding. If not, open a PR that updates the version in version.txt and merge it in before proceeding with the following steps. For example, if OSB is currently on version 0.3.0 and we want to release the next version as 0.4.0, update version.txt from 0.3.0 to 0.4.0.
+* Ensure you have git cloned the official OpenSearch Benchmark repository with the ssh address on your local computer.
+* Ensure that all new committed changes in OSB that are visible by users are added to documentation
+
+## Release new version of OpenSearch Benchmark to PyPi, Docker, and ECR
+
+NOTE: The version number below is in semantic format, for instance, 1.2.0.
+
+1. Create a tag: `git tag <VERSION> main`
+    1. Ensure that this is done in the main official opensearch-benchmark repository
+    2. This should be the new version that you have in version.txt.
+2. Push the tag: `git push origin <VERSION>`
+    1. This starts a workflow in Jenkins and creates an automated issue in the OSB repository. The issue needs to be commented on by a maintainer of the repository for the release process to proceed.
+    2. Example of automated issue opened by Jenkins Workflow
+3. Maintainer needs to comment on Automated Issue: Once Maintainer has commented, the workflow uploads OSB to PyPi and OSB Dockerhub Staging account. Once the workflows are finished publishing OSB to PyPI and OSB Dockerhub staging account (verify here), person who pushed the tag should visit both PyPi and Dockerhub staging.
```

Comment:
Once the maintainer has commented,
uploads OSB to the PyPI and OSB..
the person who pushed the tag
DockerHub staging to verify that the artifacts have been uploaded.